src/etc/init.d/S50dropbear

```diff
@@ -13,25 +13,33 @@ start() {
 
 	echo -n "Starting dropbear sshd: "
 	umask 077
-	# Ensure host keys are changed when instance ID changes
-	cirros-per instance remove-dropbear-host-keys -- rm -rf /etc/dropbear
-	
-	# Make sure dropbear directory exists
-	if [ ! -d /etc/dropbear ]; then
-		mkdir -p /etc/dropbear
+	local dropbear_key_dir="/etc/dropbear"
+
+	# Handle symlinked directories
+	if [ -L "$dropbear_key_dir" ]
+	then
+		dropbear_key_dir=$(readlink "$dropbear_key_dir")
+	fi
+
+	if [ -d "$dropbear_key_dir" ]
```

Comment:
You actually do not want to wrap this in an `if` check.
We want `cirros-per` to run once per instance independent of whether or not that is a directory. 
The way you have it now, cirros-per instance would not be called unless that directory exists.

So if on the first boot it did *not* exist, then we wouldn't clear out the keys *or* lay down the
marker that said we had.

then, on the second boot, the directory would exist, but the marker wouldn't, so we'd wipe the keys.

Reply:
Yeah, I wasn't sure about that part. So the code should look like this?
```
# Ensure host keys are changed when instance ID changes
cirros-per instance remove-dropbear-host-keys -- rm -rf "$dropbear_key_dir"
mkdir -p "$dropbear_key_dir"
```

Reply:
Do we need the symlink? Do we want to retain it?

What was the original symlink to?

These are things I probably should know, but do not.

Reply:
So this commit sheds some light on the problem [1]. The change was to handle RO filesystems. AFAICT CirrOS's root tries to remount the root with 'rw' just after init [2], but I don't see any hard failures if that doesn't succeed. If RO root is not supported by CirrOS it seems we can simply remove the symlink and replace it with a regular dir.

[1] https://github.com/buildroot/buildroot/commit/e7d04dd2df8bb935c61f7c814ee88eba7e75b5e4
[2] https://github.com/cirros-dev/cirros/blob/deb183350e164ad7f6169b6285697661f7f3f22d/src/etc/init.d/rc.sysinit#L36

Reply:
Where is the RW link point to?  I'm fine with moving towards containing writable files/directories.

How about making this into two pull requests.

One trivial one to just drop the dss. Open another one for that, and we'll land it quickly.
Then use this one for symlink /rw changes.

thanks.

Reply:
> One trivial one to just drop the dss

[Done](https://github.com/cirros-dev/cirros/pull/46)

> Where is the RW link point to?

Should be CirrOS's 'src/etc/init.d/rc.sysinit:36', let me [retry](https://github.com/cirros-dev/cirros/blob/deb183350e164ad7f6169b6285697661f7f3f22d/src/etc/init.d/rc.sysinit#L36).